src/main/java/com/faforever/client/map/MapServiceImpl.java

```diff
@@ -66,7 +66,14 @@ public Image loadSmallPreview(String mapName) {
 
     logger.debug("Fetching small preview for map {} from {}", mapName, url);
 
-    return new Image(url, true);
+    Image image = new Image(url, true);
+    if (image.isError()) {
+      logger.warn("Error fetching small preview for map {} from {}", mapName, url);
+      String theme = preferencesService.getPreferences().getTheme();
+      return new Image(ThemeUtil.themeFile(theme, "images/map_background.png"));
+    } else {
+      return image;
+    }
```

Comment:
This can't reliably work; `new Image(url, true)` loads the image in background (that's what the `true` stands for) so at the time of `if(image.isError())` you'll probably always get false.